buildtest/cli/__init__.py

```diff
@@ -1027,6 +1021,11 @@ def config_menu(subparsers, parent_parser):
     compiler_test.add_argument(
         "compiler_names", nargs="*", help="Specify compiler name to test"
     )
+    # use parent parser for defining --file option
+    file_parser = argparse.ArgumentParser(parents=[parent_parser])
+    file_parser.add_argument(
+        "--file", action="store_true", help="Wrtie configuration file to a new file"
+    )
```

Comment:
can you please define the `file_parser` next to https://github.com/buildtesters/buildtest/blob/ff6de89df077fad36df6e212d45d296843d5edd2/buildtest/cli/__init__.py#L226 I think we should probably have some variable `self.parent_parsers` as a list and then we can pass this around to each method where needed.

Have something like this 

```
self.parent_parsers['file'] = argparse.ArgumentParser(add_help=False)
self.parent_parsers['file'].add_argument("--file", action="store_true", help="Wrtie configuration file to a new file")
```

Then you can add the parent parser in the following line

https://github.com/buildtesters/buildtest/blob/ff6de89df077fad36df6e212d45d296843d5edd2/buildtest/cli/__init__.py#L993-L996

by doing 

```
    compiler_find = subparsers_compiler.add_parser(
        "find",
        help="Find compilers",
        parents=[self.parent_parsers['file']]
    )
```

Does that make sense?.

Reply:
I assume I will have the following in the get_parser function:

parent_parser = {}
parent_parser["pager"] = argparse.ArgumentParser(add_help=False)
parent_parser["pager"].add_argument(
        "--pager", action="store_true", help="Enable PAGING when viewing result"
    )
parent_parser["file"] = argparse.ArgumentParser(add_help=False)
parent_parser["file"].add_argument(
        "--file", action="store_true", help="Wrtie configuration file to a new file"

And then pass parent_parser around to each menu method where needed. 

If my understanding is correct, only in compiler_find will parent = [self.parent_parsers['file']].
All others will become parent = [self.parent_parsers["pager"]]? 

Thank you!!

Reply:
yes except change `parent_parser` to `self.parent_parser` that way we dont need to pass variable around each method. Yes i think you understand the logic. 

Depending on the subcommand you will have `parents=[self.parent_parsers['pager']` where `--pager` option is inherited. Likewise, where `--file` is inherited you can do `parents=[self.parent_parsers['file']]`. In event multiple arguments need to be inherited to a sub-command we should be able to index the dict so `parents=[self.parent_parsers['pager'], [self.parent_parsers['file']]`.

Also a tip when copying code in markdown you can use triple backticks to format code. If you want to format python code you can do ```python. Github supports syntax highlighting for several languages you should take a look at https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting